service/history/execution/integrity.go

```diff
@@ -62,7 +58,7 @@ func GetResurrectedTimers(
 	iter := collection.NewPagingIterator(getHistoryPaginationFn(
 		ctx,
 		shard,
-		minTimerStartedID,
+		1,
```

Comment:
ah, yeah, I guess this would have missed things that were lower ID somehow.  or if it entirely lost a timer.

makes sense 👍

Reply:
This actually can include invalid nodes. For example, the history branch may look like this 
```
node_id, txn_id
------------------
1,             10
2,             20 (contains event 2 and 3)
2,             11
3,             12
4,             21
```
node_id =3 is an invalid node, because the txn_id is lower than previous node. But if we read events from 3, that node is included, because we don't know the previous txn_id and do not discard that node.